GDJS/Runtime/RuntimeInstanceContainer.ts

```diff
@@ -0,0 +1,696 @@
+
+    /** Contains the objects data stored in the project */
+    _objects: Hashtable<ObjectData>;
+    _objectsCtor: Hashtable<typeof RuntimeObject>;
+
+    _layers: Hashtable<Layer>;
+    _layersCameraCoordinates: Record<string, [float, float, float, float]> = {};
+
+    // Options for the debug draw:
+    _debugDrawEnabled: boolean = false;
+    _debugDrawShowHiddenInstances: boolean = false;
+    _debugDrawShowPointsNames: boolean = false;
+    _debugDrawShowCustomPoints: boolean = false;
+
+    constructor() {
+      this._instances = new Hashtable();
+      this._instancesCache = new Hashtable();
+      this._objects = new Hashtable();
+      this._objectsCtor = new Hashtable();
+      this._layers = new Hashtable();
+    }
+
+    /**
+     * Return the time elapsed since the last frame,
+     * in milliseconds, for objects on the layer.
+     */
+    abstract getElapsedTime(): float;
+
+    /**
+     * Get the renderer associated to the container.
+     */
+    abstract getRenderer(): gdjs.RuntimeInstanceContainerRenderer;
+
+    /**
+     * Get the renderer for visual debugging associated to the container.
+     */
+    abstract getDebuggerRenderer(): gdjs.DebuggerRenderer;
+
+    /**
+     * Get the {@link gdjs.RuntimeGame} associated to this.
+     */
+    abstract getGame(): gdjs.RuntimeGame;
+
+    /**
+     * Get the {@link gdjs.RuntimeScene} associated to this.
+     */
+    abstract getScene(): gdjs.RuntimeScene;
+
+    /**
+     * Convert a point from the canvas coordinates (for example,
+     * the mouse position) to the container coordinates.
+     *
+     * @param x The x position, in container coordinates.
+     * @param y The y position, in container coordinates.
+     * @param result The point instance that is used to return the result.
+     */
+    abstract convertCoords(x: float, y: float, result?: FloatPoint): FloatPoint;
+
+    /**
+     * Convert a point from the container coordinates (for example,
+     * an object position) to the canvas coordinates.
+     *
+     * @param sceneX The x position, in container coordinates.
+     * @param sceneY The y position, in container coordinates.
+     * @param result The point instance that is used to return the result.
+     */
+    abstract convertInverseCoords(
+      sceneX: float,
+      sceneY: float,
+      result: FloatPoint
+    ): FloatPoint;
+
+    /**
+     * @return the width of the game resolution for a {@link gdjs.RuntimeScene} or the
+     * internal size for a {@link gdjs.CustomRuntimeObject}.
+     */
+    abstract getViewportWidth(): float;
+
+    /**
+     * @return the height of the game resolution for a {@link gdjs.RuntimeScene} or the
+     * internal size for a {@link gdjs.CustomRuntimeObject}.
+     */
+    abstract getViewportHeight(): float;
+
+    /**
+     * @return the center X of the game resolution for a {@link gdjs.RuntimeScene} or the
+     * internal size for a {@link gdjs.CustomRuntimeObject}.
```

Comment:
Let's write down your explanation:

```suggestion
     * internal size for a {@link gdjs.CustomRuntimeObject} (AABB of all children for the default dimension of the object).
```